### PR TITLE
Compute total bytes in Body constructor when possible

### PR DESCRIPTION
**Pull request by tekwiz**
**What is the purpose of this pull request?**

- [ ] Documentation update
- [x] Bug fix
- [ ] New feature
- [ ] Other, please explain:

**What changes did you make? (provide an overview)**

When a `Body` is constructed from something other than a stream, it now computes the `totalBytes` in the constructor instead of with `getTotalBytes`.  This fixes a few issues with the construction of bodies from `FormData` objects.

This change also sets the 

**Which issue (if any) does this pull request address?**

#1045

**Is there anything you'd like reviewers to know?**

no

### CODE REVIEW

**Review by xxczaki — APPROVED** [review]

**Review by wh0 — APPROVED** [review]
oops, long delayed review, but this looks good. thanks!

File: test/main.js [hunk 1]
```diff
@@ -2127,6 +2128,31 @@ describe('node-fetch', () => {
...
+			method: 'POST',
+			body: formBody,
+			size: 1024
+		});
+		// override the body to be the original formBody object
+		Object.defineProperty(formRequest, 'body', {value: formBody});
+
+		expect(getTotalBytes(formRequest)).to.equal(141);
+		expect(extractContentType(formBody, formRequest)).to.startWith('multipart/form-data');
+	});
```
Comment (jimmywarting) [inline test/main.js:2154]: This isn't really a spec compatible FormData
the most important part is the iterator

Comment (wh0) [reply test/main.js:2154]: Map supports iteration

**Comment by jimmywarting** [comment]
bump @tekwiz

**Comment by tekwiz** [comment]
Superseded by #1314

**Pull request closed without merging**